cmake/sysbuild/sdp.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Update FLPR image KConfig based on main app configuration.
+#
+# Usage:
+#   flpr_egpio_update_kconfig()
+#
+function(flpr_egpio_update_kconfig)
+
+  sysbuild_get(CONFIG_MBOX_BACKEND IMAGE ${DEFAULT_IMAGE} VAR CONFIG_GPIO_NRFE_EGPIO_BACKEND_MBOX KCONFIG)
+  sysbuild_get(CONFIG_ICMSG_BACKEND IMAGE ${DEFAULT_IMAGE} VAR CONFIG_GPIO_NRFE_EGPIO_BACKEND_ICMSG KCONFIG)
+  sysbuild_get(CONFIG_ICBMSG_BACKEND IMAGE ${DEFAULT_IMAGE} VAR CONFIG_GPIO_NRFE_EGPIO_BACKEND_ICBMSG KCONFIG)
+  sysbuild_get(CONFIG_PLATFORM IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SOC KCONFIG)
+
+  message(STATUS "eGPIO: Using ${CONFIG_PLATFORM} SoC")
+
+  if(CONFIG_MBOX_BACKEND)
+    sysbuild_cache_set(VAR egpio_CONFIG_GPIO_NRFE_EGPIO_BACKEND_MBOX APPEND REMOVE_DUPLICATES "y")
+    if(CONFIG_PLATFORM STREQUAL "nrf54l15")
```

Comment:
use SoC Kconfig symbol not comparing strings

Reply:
Added in #16592, CC: @jaz1-nordic